Introduction to Artificial Intelligence and Machine Learning (#425)
build-on-aws/content: posts/introduction-to-artificial-intelligence-and-machine-learning/index.md

For example, Fig. 5 shows a simple example of how values of a property increase as its surface area increases, creating a linear relationship between both characteristics.

![Example of Regression: Relationship between surface of a property and its value](images/fig_13.png) <h4 align="center">Fig. 5. Regression Example: Relationship between superficies of a property and its value.</h4> 

- **1.2. Classification:** Used to assign data to one of several categories (classes) based on certain characteristics. Once classifier has been altered, it can be used to make predictions about new and unknown data.

Binary classifiers deliver a value between 0 and 1 and multi-class classifiers capable of delivering a list of probability between 0 and 1, one for each class (our example of animals in Deep Learning).

2. **Unsupervised Learning:** The labels are not known, the model builds the patterns and structure of the data. It is divided into two types:

> - **2.1. Clusterization**: Is a technique used to divide a data set into groups (clusters) according to certain characteristics, so data in same cluster is similar to each other, while data in different clusters is different.

Reviewer: ```suggestion
- **2.1. Clusterization**: Is a technique used to divide a data set into groups (clusters) based on certain characteristics. Data in the same cluster is more similar to each other than data in different clusters.
```

3. **Reinforcement learning:** it is a technique (Fig. 7) in which an agent interacts with its environment and receives rewards or punishments based on its actions. Agent learns through exploration and experimentation, striving to maximize the total reward received over time. This is accomplished by selecting actions that agent believes are most likely to produce a reward.

One example is [AWS Deep Racer](https://aws.amazon.com/deepracer/), where models are trained to compete in races as cars within tracks (virtual or physical).

![Reinforcement learning Example](images/fig_15.png) <h4 align="center">Fig. 7. Reinforcement Learning example. </h4> 

Now that we are more clear about what AI/ML is and how it works, we can approach an ML project by identifying phases in ML life cycle.

## Phases of life cycle of a Machine Learning project.

![Phases of life cycle of a Machine Learning project](images/fig_16.png) <h4 align="center">Fig. 8. Phases of life cycle of a Machine Learning project. </h4> 

> ML life cycle is a iterative and cyclical process (Fig. 8), adds clarity, insight into entire process, and structure to maximize  success of a ML project.

Reviewer: ```suggestion
The ML life cycle is an iterative and cyclical process (as depicted in Fig. 8), which provides clarity and insight into the entire process, structuring it to maximize the success of an ML project.
```

> The best way to learn is with examples, putting your hands in, but in this blog we don’t have code to “get dirty”, so let’s develop phases assuming we want to build a model that evaluates bank fraud:

Reviewer: ```suggestion
The best way to learn is through examples and hands-on practice. While this blog doesn't provide code to "get your hands dirty", let's explore the phases of this process by assuming we want to build a model to detect bank fraud:
```

|  **Monitoring:** We define rules to detect problems and send alerts. Problems to detect in this phase include: data and model quality, bias deviation among others. If necessary,  model should be re-trained. | Check whether data that is coming to us is statistically similar to ones that trained the model, we must also deliver inferences and the final result of  transaction in form of reports, or visualization panels (that is, It is important to confirm that  project is meeting objectives). Another action at this stage is to use experts to review doubtful transactions to help effectively define level of fraud and feed back model with new data for re-training.|

---

Most important thing about these models (apart from having excellent performance) is that the people who use it believe in it, for example, what would happen if a customer is making a legal purchase and the model labels it fraudulent by blocking card? we would have an angry customer wanting to change banks.

> Always think in your client, how to provide them with best service, how to improve their quality of life or simply, how to make them happy, because client is not necessarily a person from whom you receive a monetary reward or an area in your organization; A client can also be your family who will enjoy some new application that you as a developer, now with knowledge in AI / ML, created for your home.

Reviewer: ```suggestion
Always think about your client, how to provide them with best service, how to improve their quality of life or simply, how to make them happy, because client is not necessarily a person from whom you receive a monetary reward or an area in your organization; A client can also be your family who will enjoy some new application that you as a developer, now with knowledge in AI / ML, created for your home.
```

## Conclusions

I hope you enjoyed this read and it made you curious to develop more products based on AI/ML, now that you know the answers to the questions: What is Artificial Intelligence? What is machine learning? and How does an ML model learn?

> We delve into concept of ML, the types that are learned and subset of ML called Deep Learning and at the end we develop the phases of an ML project with an example of bank fraud.

Reviewer: ```suggestion
We've delved into the concept of ML, explored the types of learning, delved into a subset of ML known as Deep Learning, and finally, walked through the phases of an ML project using a bank fraud detection example.
```
